### ISSUE TITLE
Unconditioned Flowtron

### ISSUE THREAD

**Issue by adrianastan** [NVIDIA/flowtron#47]
Hi,

Did anybody try to train the Flowtron flow architecture in an unconditioned manner, for density estimation for example? If so, any hints and tips you could share?

Thanks!

**Comment by rafaelvalle** [comment]
Train a model with 1 step of flow first. 
Then use this model to warm-start a model with 2 steps of flow.

**Comment by adrianastan** [comment]
Hi,

Thanks for your reply. I indeed started training a 1-flow using the LibriSpeech  train-clean-100 data using a modified unconditioned [version of Flowtron](https://gitlab.utcluj.ro/sadriana/flowtron-librispeech). I then used the trained flow to warm-start a 2-flow architecture. However at inference there is nothing but noise: https://drive.google.com/file/d/1V7sX3Ma3RFBo6lNSCUxSsNjP3Y_HmAZo/view?usp=sharing. 

![sid0_sigma0 5](https://user-images.githubusercontent.com/6659449/88794664-7b80be80-d1a7-11ea-8d55-18c00f9763d3.png)

I was expecting at least some babble noise.

Any hints on when is a goot point to start the second-flow training? Should I train more? Should I lower the learning rate?
Below are the loss curves for the 1st flow:

![Screenshot from 2020-07-29 14-18-24](https://user-images.githubusercontent.com/6659449/88794627-6dcb3900-d1a7-11ea-9824-29940e8882cd.png)

Thanks!

**Comment by rafaelvalle** [comment]
The validation loss for your 1-step of flow model is starting to plateau. 
Use this model to warm-start a 2-steps of flow model. I assume the validation loss will go down.
You can alternatively try the same experiment on LJS.

**Comment by adrianastan** [comment]
I warmstarted a 2 flow model from the 1 flow weights and continued training. Training and validation losses are as below:


![2flows](https://user-images.githubusercontent.com/6659449/89275929-a0fc4500-d64b-11ea-8377-fa6d3f922d15.png)

![2flows_sid0_sigma0 5](https://user-images.githubusercontent.com/6659449/89275966-afe2f780-d64b-11ea-86d3-89dff3cf0dc3.png)

Still no speech-like output at inference. 
https://drive.google.com/file/d/19OC2cSfPgfvrS0mrRx73bkLLKp0yt0v8/view?usp=sharing

I additionally started a subsequent 3 flow model, as well:

![3flows](https://user-images.githubusercontent.com/6659449/89275940-a6598f80-d64b-11ea-81f2-b9329b26db73.png)

The output is as follows:

![3flows_sid0_sigma0 5](https://user-images.githubusercontent.com/6659449/89276000-bb362300-d64b-11ea-9fe9-5e5f05f655ea.png)

https://drive.google.com/file/d/1F7lXcEqx5_gqMDog4KgyahfKDGx7-175/view?usp=sharing


So I assume that this architecture might not be complex enough to estimate a multispeaker latent space. I will try to do the same thing on LJSpeech -- perhaps the conditions are simpler. 

Thanks!

**Comment by rafaelvalle** [comment]
@adrianastan if you trained a model with speaker embeddings, what happens if do this:
```flowtron.infer(flowtron.forward(audio, speaker), other_speaker)```

**Comment by adrianastan** [comment]
I did not use speaker embeddings, just a multispeaker dataset. I removed all conditionings of the flow.
